load_data.py

```diff
@@ -65,17 +79,22 @@ def data_loader(dataset, file_name, dtype):
         with open(str(DATA_DIR + '/cifar-10-batches-py/' + file_name), 'rb') as fo:
             data_dict = cPickle.load(fo)
 
-        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype))
+        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype),
+                                      dtype=dtype,
+                                      use_grayscale=use_grayscale
+                                      )
         labels = to_categorical(np.asarray(data_dict['labels'], dtype=dtype),
                 num_classes=10)
 
     elif dataset == "cifar100":
         with open(str(DATA_DIR + '/cifar-100-python/' + file_name), 'rb') as fo:
             data_dict = cPickle.load(fo)
 
-        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype))
+        data = preprocess_cifar_imgs_(np.asarray(data_dict['data'], dtype=dtype),
+                                      dtype=dtype,
```

Comment:
What about this Is this ```dtype=dtype```; is that also necessary?

Reply:
Same, necessary for the conversion to grayscale